include/bluetooth/mesh/vnd/dm_cli.h

```diff
@@ -0,0 +1,221 @@
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_dm_cli Distance Measurement Client model
+ * @{
+ * @brief API for the Distance Measurement Client model.
+ */
+
+#ifndef BT_MESH_DM_CLI_H__
+#define BT_MESH_DM_CLI_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <bluetooth/mesh/models.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/bluetooth/mesh/access.h>
+#include "dm_common.h"
+#include <dm.h>
+
+/** Parameters for the Distance measurement result status message. */
+struct bt_mesh_dm_cli_result_status {
+	/** Message status code. */
+	uint8_t status;
+	/** Number of entries in the result. */
+	uint8_t entry_cnt;
+	/** Result memory object. */
+	struct bt_mesh_dm_res_entry *res;
+};
+
+/** Parameters for the Distance measurement config status message. */
+struct bt_mesh_dm_cli_cfg_status {
+	/** Message status code. */
+	uint8_t status;
+	/** Measurement in progress flag. */
+	bool is_in_progress;
+	/** Number of result entries available. */
+	uint8_t result_entry_cnt;
+	/** Default parameter values. */
+	struct bt_mesh_dm_cfg def;
+};
+
+/** Parameters for the Distance measurement config start message. */
+struct bt_mesh_dm_cli_cfg_start {
+	/** Measurement method. */
+	enum dm_ranging_mode method;
+	/** Target unicast address. */
+	uint16_t addr;
+	/** Default parameter values. */
+	struct bt_mesh_dm_cfg *def;
+};
+
+/** Handler functions for the Distance Measurement Client. */
+struct bt_mesh_dm_cli_handlers {
+	/** @brief Config status message handler.
+	 *
+	 * @note This handler is optional.
+	 *
+	 * @param[in] cli Client that received the status message.
+	 * @param[in] ctx Context of the message.
+	 * @param[in] status Config status contained in the message.
+	 */
+	void (*const cfg_status_handler)(struct bt_mesh_dm_cli *cli,
+					 struct bt_mesh_msg_ctx *ctx,
+					 const struct bt_mesh_dm_cli_cfg_status *status);
+
+	/** @brief Result status message handler.
+	 *
+	 * @note This handler is optional.
+	 *
+	 * @param[in] cli Client that received the status message.
+	 * @param[in] ctx Context of the message.
+	 * @param[in] result Result contained in the message.
+	 */
+	void (*const result_handler)(struct bt_mesh_dm_cli *cli,
+				     struct bt_mesh_msg_ctx *ctx,
+				     const struct bt_mesh_dm_cli_result_status *result);
+};
+
+/** Distance Measurement Client instance. */
+struct bt_mesh_dm_cli {
+	/** Application handler functions. */
+	const struct bt_mesh_dm_cli_handlers *const handlers;
+	/** Current Transaction ID. */
+	uint8_t tid;
+	/** Access model pointer. */
+	struct bt_mesh_model *model;
+	/** Response context for tracking acknowledged messages. */
+	struct bt_mesh_msg_ack_ctx ack_ctx;
+	/** Publish parameters. */
+	struct bt_mesh_model_pub pub;
+	/* Publication buffer */
+	struct net_buf_simple pub_buf;
+	/* Publication data */
+	uint8_t pub_data[BT_MESH_MODEL_BUF_LEN(BT_MESH_DM_START_OP,
```

Comment:
To be honest I keept this here since it seems the client is not able to send anything without this parameter being initialized in the init CB. Do you have an example of a model where these parameters are omited? Otherwise im not sure how I can make it work.

Reply:
https://github.com/balaklaka/sdk-nrf/blob/ac44f76f01a69a594ee736d098ef50b5c10db8c7/subsys/bluetooth/mesh/gen_onoff_cli.c#L117

Basically, if publication is not available for the model user must always provide 'ctx' parameter to the client API.

Reply:
Ok, so I guess that we can just keep it as it is then?

Reply:
What I mean, it is best to remove the publication and enforce in the client API that "ctx" is not NULL. If the publication is kept enabled then potentially one can configure DM client to periodically publish at 100 ms intervals and that will lead to broken behavior. There is no use case where publication is absolutely needed for DM clients.